packages/network/__tests__/fetch-test.ts

```diff
@@ -82,4 +119,149 @@ describe('@data-eden/fetch', function () {
+            "headers": {
+              "accept": "*/*",
+              "accept-encoding": "gzip,deflate",
+              "connection": "close",
+              "host": "www.example.com",
+              "user-agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
+              "x-csrf": "a totally legit request",
+            },
+            "names": {},
+          },
+          "status": "success",
+        }
+      `);
+  });
+
+  test('should be able to change the http method for a given request', async () => {
+    expect.assertions(2);
+
+    const queryTunneling: Middleware = async (
+      request: Request,
+      next: NormalizedFetch
+    ) => {
+      if (request.url.length <= 10) {
+        // no tunneling needed
+        return next(request);
+      }
+
+      const url = new URL(request.url);
+      request.headers.set('X-HTTP-Method-Override', request.method);
+      const tunneledRequest = new Request(
+        `${url.protocol}//${url.hostname}${url.pathname}`,
+        {
+          method: 'POST',
+          headers: request.headers,
+          body: url.searchParams,
+        }
+      );
+
+      return next(tunneledRequest);
+    };
+
+    const fetch = buildFetch([csrfMiddleware, queryTunneling, noopMiddleware]);
+
+    const response = await fetch('http://www.example.com/resource/preview');
+
+    expect(response.status).toEqual(200);
+    expect(await response.json()).toMatchInlineSnapshot(`
+      {
+        "headers": {
+          "headers": {
+            "accept": "*/*",
+            "accept-encoding": "gzip,deflate",
+            "connection": "close",
+            "content-length": "0",
+            "content-type": "application/x-www-form-urlencoded;charset=UTF-8",
+            "host": "www.example.com",
+            "user-agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
+            "x-csrf": "a totally legit request",
+            "x-http-method-override": "GET",
+          },
+          "names": {},
+        },
+        "method": "POST",
+        "status": "success",
+      }
+    `);
+  });
+
+  test('should be able to maintain order of middlewares passed', async () => {
+    expect.assertions(6);
+
+    const middlewareOne: Middleware = async (
+      request: Request,
+      next: NormalizedFetch
+    ) => {
+      expect(request.headers.get('two')).toBeFalsy();
+      expect(request.headers.get('three')).toBeFalsy();
+
+      return next(request);
+    };
+
+    const middlewareTwo: Middleware = async (
+      request: Request,
+      next: NormalizedFetch
+    ) => {
+      expect(request.headers.get('three')).toBeFalsy();
+
+      request.headers.set('two', 'true');
+
+      return next(request);
+    };
+
+    const middlewareThree: Middleware = async (
+      request: Request,
+      next: NormalizedFetch
+    ) => {
+      expect(request.headers.get('two')).toEqual('true');
+
+      request.headers.set('three', 'true');
+
+      return next(request);
+    };
```

Comment:
Instead of asserting the "negative" (e.g. that the headers that will be added by another middleware are not populated yet), I'd love to assert the full state of `request.header` (something like `expect(headers).toMatchInlineSnapshot()`) instead.

What do you think?